zbar_ros/src/barcode_reader_node.cpp

```diff
@@ -74,14 +74,29 @@ void BarcodeReaderNode::imageCb(sensor_msgs::msg::Image::ConstSharedPtr image)
   auto it_end = zbar_image.symbol_end();
   if (it_start != it_end) {
     // If there are barcodes in the image, iterate over all barcode readings from image
-    for (zbar::Image::SymbolIterator symbol = it_start; symbol != it_end; ++symbol) {
-      std::string barcode = symbol->get_data();
-      RCLCPP_DEBUG(get_logger(), "Barcode detected with data: '%s'", barcode.c_str());
+    for (zbar::Image::SymbolIterator symbol_it = it_start; symbol_it != it_end; ++symbol_it) {
+      zbar_ros_interfaces::msg::Symbol symbol;
+      symbol.data = symbol_it->get_data();
+      RCLCPP_DEBUG(get_logger(), "Barcode detected with data: '%s'", symbol.data.c_str());
+
+      RCLCPP_DEBUG(
+        get_logger(), "Polygon around barcode has %d points", symbol_it->get_location_size());
```

Comment:
It's a shame vision_msgs only has 'bounding box' type boundaries rather than the polygons zbar reports.